### ISSUE TITLE
PublishMessage eligible

### ISSUE THREAD

**Issue by her-ur** [voryx/ThruwayCommon#2]
Hello, I am using eligible option when I am publishing messages. Here https://github.com/voryx/Thruway/blob/master/src/Transport/InternalClientTransportProvider.php#L49 is all message converted to `stdClass` by `json_decode(json_encode($msg))` and when is message recreated here https://github.com/voryx/ThruwayCommon/blob/master/src/Message/PublishMessage.php#L159, there is a problem with this condition
```
$this->eligible = isset($options->eligible) && is_array($options->eligible) ? $options->eligible : null;
```
because `$options->eligible` is `stdClass` and not an array, eligible property is not set and messages are then broadcasted to everyone.
I think it's bug. How that can be fixed? If you give me some details please I can submit a PR.
Thank you!

**Comment by mbonneau** [comment]
@her-ur Eligible should be an array or list of session ids, so when it get serialized, it should be converted to a JSON array.

Can you show me the code that you are using?

**Comment by her-ur** [comment]
Of course, this is called in class that extends `Thruway\Peer\Client` and is used as router module.
```
$this->getSession()->publish($topic->getValue(), [$message], [], ['eligible' => $this->sessions[$userPublicToken]]);
```
`$this->sessions[$userPublicToken]` is array of session ids. Weird is that sometimes its working fine, but when i am refreshing pages it goes wrong as I described above.
